patches/server/1030-PreSpawnerSearchEvent.patch

```diff
@@ -0,0 +1,57 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: TreemanKing <67459602+TreemanKing@users.noreply.github.com>
+Date: Wed, 16 Aug 2023 11:53:44 +1000
+Subject: [PATCH] PreSpawnerSearchEvent
+
+This patch introduces a new event, PreSpawnerSearchEvent. This event is designed to be fired before a mob spawner attempts to search for nearby players to activate. It provides the ability to control and customize the activation logic of mob spawners, preventing them from searching and spawning mobs if desired
+
+diff --git a/src/main/java/net/minecraft/world/level/BaseSpawner.java b/src/main/java/net/minecraft/world/level/BaseSpawner.java
+index 633500aefd515df5dadda3802b94079f75a03fa0..9c3ee094035e6d903d3732349c9ff1e0a93b891a 100644
+--- a/src/main/java/net/minecraft/world/level/BaseSpawner.java
++++ b/src/main/java/net/minecraft/world/level/BaseSpawner.java
+@@ -46,6 +46,7 @@ public abstract class BaseSpawner {
+     public int requiredPlayerRange = 16;
+     public int spawnRange = 4;
+     private int tickDelay = 0; // Paper
++    private int searchCooldownTicks = 0; // Paper
+ 
+     public BaseSpawner() {}
+ 
+@@ -54,7 +55,7 @@ public abstract class BaseSpawner {
+         this.spawnPotentials = SimpleWeightedRandomList.empty(); // CraftBukkit - SPIGOT-3496, MC-92282
+     }
+ 
+-    public boolean isNearPlayer(Level world, BlockPos pos) {
++    public boolean isNearPlayer(Level world, BlockPos pos) { // diff on change - PreSpawnerSearchEvent
```

Comment:
The diff on change should be on the hasNearbyAlivePlayerThatAffectsSpawning line, you can merge both comments in a single one to avoid duplicate // Paper